src/docs/testing/best-practices.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Performance best practices
+short-title: Best practices
+description: How to ensure that your Flutter app is performant.
+---
+
+Generally, Flutter applications are performant by default,
+so you only need to avoid common pitfalls to get excellent performance
+instead of needing to micro-optimize with complicated profiling tools.
+These best recommendations will help you write the most performant
+Flutter app possible.
+
+## Best practices
+
+How do you design a Flutter app to most efficiently render your scenes?
+In particular, how do you ensure that the painting code generated by the
+framework is as efficient as possible? Here are a few things to consider
+when designing your app:
+
+### Call setState thoughtfully
+
+* Avoid repetitive and costly work in `build()` methods since `build()`
+  can be invoked frequently when ancestor Widgets rebuild.
+* Localize the `setState()` call to the part of the subtree whose UI
+  actually needs to change. Avoid calling setState() high up in
+  the tree if the change is contained to a small part of the tree.
+* When a widget rebuilds, but the build stops a child from rebuilding,
+  reuse the same child widget.
+
+Also see:
+
+* [Performance considerations](https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/StatefulWidget-class.html#performance-considerations),
+  part of the
+  [StatefulWidget](https://master-docs-flutter-io.firebaseapp.com/flutter/widgets/StatefulWidget-class.html)
+  API doc
+* [Improve your Flutter app performance: split your widgets
```

Comment:
I wouldn't necessarily highlight this. There's nothing wrong with it but it's a very particular and opinionated choice out of many many choices the developer can individually make. I would just remove this reference.

Reply:
Done. Thanks, @xster!